src/platform/datapath_raw_socket.c

```diff
@@ -307,10 +301,28 @@ CxPlatDpRawParseEthernet(
         return;
     }
 
-    if (Ethernet->Type == 0x0008) { // IPv4
+    uint16_t EthernetType = QuicNetByteSwapShort(Ethernet->Type);
```

Comment:
IMO, the byteswap is unnecessary. Just compare in wire format. I know it's a micro-optimization, but it's a trivial one to do. With all these PRs, we should keep perf at the forefront of our minds.